utils/utils_daemon.go

```diff
@@ -3,14 +3,14 @@
 package utils
 
 import (
+	"github.com/docker/docker/pkg/system"
 	"os"
-	"syscall"
 )
 
 // IsFileOwner checks whether the current user is the owner of the given file.
 func IsFileOwner(f string) bool {
-	if fileInfo, err := os.Stat(f); err == nil && fileInfo != nil {
-		if stat, ok := fileInfo.Sys().(*syscall.Stat_t); ok && int(stat.Uid) == os.Getuid() {
+	if fileInfo, err := system.Lstat(f); err == nil && fileInfo != nil {
```

Comment:
I'm not familiar with the use of this function, but this diff be an issue considering that `system.Lstat` is semantically different than `os.Stat` for symbolic links?